### PR TITLE
feat: add Jest unit testing library

### PR DESCRIPTION
**Pull request by krisraven**
Includes for #1 :
- add Jest Types
- add Jest config with default values
- add Husky to run pre-push hook
- add test running to npm Scripts

### CODE REVIEW

**Review by jolo-dev — CHANGES_REQUESTED** [review]
Thank you @krisraven ,
Much appreciated.
However, we would like to use [PNPM](https://pnpm.io/). Could you adjust that?

**Comment by krisraven** [comment]
@jolo-dev sure, I will give it a try.

**Comment by krisraven** [comment]
Hi @jolo-dev. The `pnpm-lock.yaml` has been updated by running `pnpm install`. Is this what you were asking for in your comment, or are there other things that need to be done?

File: package.json [hunk 1]
```diff
@@ -4,16 +4,25 @@
   "description": "",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "coverage": "npm run test -- --coverage --silent",
```
Comment (jolo-dev) [inline package.json]:
```suggestion
    "coverage": "pnpm run test -- --coverage --silent",
```

File: package.json [hunk 2]
```diff
@@ -4,16 +4,25 @@
...
+    "@types/jest": "^27.0.2",
     "eslint": "^7.32.0",
     "eslint-config-prettier": "^8.3.0",
+    "husky": "^7.0.2",
+    "jest": "^27.2.4",
     "prettier": "^2.4.1"
+  },
+  "husky": {
+    "hooks": {
+      "pre-push": "npm run test"
```
Comment (jolo-dev) [inline package.json]:
```suggestion
      "pre-push": "pnpm run test"
```

**Review by jolo-dev — CHANGES_REQUESTED** [review]
@krisraven Yes, almost.
Please adjust the `package.json` or use my suggestions and delete `package-lock.json`
And then it would be ready to merge.
Thank you 👍

File: package.json (hunk 1 above)
Comment (krisraven) [reply package.json]: OK, thank you for confirming.

File: jest.config.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+module.exports = {
+	clearMocks: true,
+	restoreMocks: true,
+	resetModules: true,
+	testEnvironment: "node",
+	testPathIgnorePatterns: ["<rootDir>/node_modules", "<rootDir>/monorepo-lambda"],
```
Comment (jolo-dev) [inline jest.config.js]: We don't need to test the `typescript-lambda` as well.

```suggestion
	testPathIgnorePatterns: ["<rootDir>/node_modules", "<rootDir>/monorepo-lambda", "<rootDir>/typescript-lambda"],
```

**Review by jolo-dev — CHANGES_REQUESTED** [review]
Awesome. We are almost there.
We still don't need the `package-lock.json`. Please delete that.

**Comment by krisraven** [comment]
Apologies for the late reply @jolo-dev. Got caught up with attending to some other things. Thanks for fixing and merging. 🏁

**Comment by jolo-dev** [comment]
All good. You did most of the job. Thank you :)

**Pull request merged**